tests/Microsoft.Identity.Test.Integration.net45/HeadlessTests/ConfidentialClientIntegrationTests.cs

```diff
@@ -0,0 +1,86 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net;
+using System.Security;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Identity.Client;
+using Microsoft.Identity.Test.Integration.Infrastructure;
+using Microsoft.Identity.Test.LabInfrastructure;
+using Microsoft.Identity.Test.Unit;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+
+namespace Microsoft.Identity.Test.Integration.net45.HeadlessTests
+{
+    [TestClass]
+    public class ConfidentialClientIntegrationTests
+    {
+        private static readonly string[] s_scopes = { "User.Read" };
+        private static readonly string[] s_oboServiceScope = { "api://23c64cd8-21e4-41dd-9756-ab9e2c23f58c/access_as_user" };
+
+        [ClassInitialize]
+        public static void ClassInitialize(TestContext context)
+        {
+            ServicePointManager.SecurityProtocol = SecurityProtocolType.Tls12;
+        }
+
+        [TestMethod]
+        public async Task ClientSecretAuthenticationAsync()
+        {
+            var keyvault = new KeyVaultSecretsProvider();
+            var secret = keyvault.GetSecret("https://buildautomation.vault.azure.net/secrets/AzureADIdentityDivisionTestAgentSecret/e360740b3411452b887e6c3097cb1037").Value;
+            var confidentialClientID = "16dab2ba-145d-4b1b-8569-bf4b9aed4dc8";
+            var confidentialClientAuthority = "https://login.windows.net/72f988bf-86f1-41af-91ab-2d7cd011db47";
```

Comment:
Can we get the clientID and tenantID from the labresponse? Not necessarily now (you can just add a TODO), but moving forward, they should provide this.

Reply:
ok, added TODO

Reply:
Any reason a todo and not added now? ETA?

---
In reply to: [279038192](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/1118#discussion_r279038192) [](ancestors = 279038192)